runtime/src/drivers.rs

```diff
@@ -171,6 +176,9 @@ impl Drivers {
             ResetReason::ColdReset => {
                 cfi_assert_eq(self.soc_ifc.reset_reason(), ResetReason::ColdReset);
                 Self::initialize_dpe(self)?;
+                if self.soc_ifc.active_mode() {
```

Comment:
I wonder if this should be a build-time flag since for a given integration it's either active or not. Otoh, it's probably easier to have a less complicated test matrix for RT.

Reply:
Would we plan on building a separate active mode ROM then?

Reply:
I think we can leave this for now. If we want to split the build, we can change this to an `if cfg!`.